src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -956,78 +1006,85 @@ async_manifest_view::time_based_retention(
-                  result.offset,
-                  result.delta);
-
-                if (!eof) {
-                    auto r = co_await cursor->next();
-                    if (
-                      r.has_value()
-                      && r.value() == async_manifest_view_cursor::eof::yes) {
-                        vlog(
-                          _ctxlog.info,
-                          "Entire archive is removed by the time-based "
-                          "retention");
-                        break;
-                    } else if (r.has_failure()) {
-                        vlog(
-                          _ctxlog.error,
-                          "Failed to scan manifest while computing retention "
-                          "{}",
-                          r.error());
-                        co_return r.as_failure();
-                    }
-                } else {
+              "Updated last offset to {}, delta {}",
+              result.offset,
+              result.delta);
+
+            if (!eof) {
+                auto r = co_await cursor->next();
+                if (
+                  r.has_value()
+                  && r.value() == async_manifest_view_cursor::eof::yes) {
                     vlog(
-                      _ctxlog.debug,
-                      "Retention found offset {} with delta {}",
-                      result.offset,
-                      result.delta);
+                      _ctxlog.info,
+                      "Entire archive is removed by the time-based "
+                      "retention");
                     break;
+                } else if (r.has_failure()) {
+                    _ctxlog.log(
+                      log_level_for_error(r.error()),
+                      "Failed to scan manifest while computing retention "
+                      "{}",
+                      r.error());
+                    co_return r.as_failure();
                 }
-            }
-            if (result.offset == model::offset{}) {
+            } else {
                 vlog(
                   _ctxlog.debug,
-                  "Failed to find the retention boundary, the manifest {} "
-                  "doesn't "
-                  "have any matching segment",
-                  cursor->manifest()->get_manifest_path());
+                  "Retention found offset {} with delta {}",
+                  result.offset,
+                  result.delta);
+                break;
             }
         }
+        if (result.offset == model::offset{}) {
+            vlog(
+              _ctxlog.debug,
+              "Failed to find the retention boundary, the manifest {} "
+              "doesn't "
+              "have any matching segment",
+              cursor->manifest()->get_manifest_path());
+        }
+    } catch (const std::system_error& err) {
+        // Thrown by `async_manifest_view::maybe_sync_manifest`
+        if (err.code() == error_outcome::shutting_down) {
```

Comment:
same problem, don't compare codes, compare `std::error_code` objects, comparison is polymorphic

Reply:
Fixed.